tidy3d/plugins/near2far/near2far.py

```diff
@@ -1,115 +1,373 @@
+        frequency: float,
+        wavelength: float,
+        pts_per_wavelength: int = 10):
+        """Returns the surface current densities associated with the surface monitor
         Parameters
         ----------
-        sim_data : :class:`.SimulationData`
-            Container for simulation data containing a near field monitor.
-        mon_name : str
-            Name of the :class:`.FieldMonitor` to use as source of near field.
-            Must be a :class:`.FieldMonitor` and stored in ``sim_data``.
         frequency : float
-            Frequency to select from the :class:`.FieldMonitor` to use for projection.
-            Must be a frequency stored in the :class:`FieldMonitor`.
+            Frequency to select from each :class:`.FieldMonitor` to use for projection.
+            Must be a frequency stored in each :class:`FieldMonitor`.
+        wavelength : float
+            Wavelength in the background medium used to calculate the number of colocation points.
+        pts_per_wavelength : int
+            Number of points per wavelength with which to discretize the
+            surface monitors for the projection. Default: 10.
         """
 
+        # make sure the monitor is a surface, i.e., exactly one of its dimensions should be zero
+        if sum(bool(size) for size in self.mon.size) != 2:
+            raise SetupError(
+                f"Can't compute far fields for the monitor {self.mon.name}; it is not a surface."
+            )
+
         try:
-            # fill nans with 0, not sure where nans come from..
-            field_data = sim_data.at_centers(mon_name).fillna(0)
+            field_data = self.sim_data[self.mon.name]
         except Exception as e:
             raise SetupError(
-                f"No data for monitor named '{mon_name}' " "found in supplied sim_data."
+                f"No data for monitor named '{self.mon.name}' " "found in supplied sim_data."
             ) from e
 
-        monitor_fields = list(field_data.keys())
-        if any(field_name not in monitor_fields for field_name in ("Ex", "Ey", "Hx", "Hy", "Hz")):
-            raise SetupError(f"Monitor named '{mon_name}' doesn't store all field values")
+        # figure out which field components are tangential or normal to the monitor
+        if self.axis == 0:
+
+            idx_uv = [1, 2]
+            tangent_fields = ['y', 'z']
+            normal_field = 'x'
+            signs = [-1.0, 1.0]
+
+        elif self.axis == 1:
+
+            idx_uv = [0, 2]
+            tangent_fields = ['x', 'z']
+            normal_field = 'y'
+            signs = [1.0, -1.0]
+
+        else:
+
+            idx_uv = [0, 1]
+            tangent_fields = ['x', 'y']
+            normal_field = 'z'
+            signs = [-1.0, 1.0]
+
+        if self.normal_dir == '-':
+            signs = [-1.0 * i for i in signs]
+
+        # compute surface current densities and delete unneeded field components
+        currents = field_data.copy(deep=True)
+
```

Comment:
let's just assign
```python
cmp_1, cmp_2 = tangent_fields
```
and use that instead of indexing `tangent_fields[i]` in many places